Source/SwiftLintFramework/Extensions/Array+SwiftLint.swift

```diff
@@ -42,4 +43,44 @@ extension Array {
             return dictionary
         }
     }
+
+    public func parallelForEach(block: @escaping ((Element) -> Void)) {
+        _ = parallelMap(transform: block)
+    }
+
+    func parallelFlatMap<T>(transform: @escaping ((Element) -> [T])) -> [T] {
+        return parallelMap(transform: transform).flatMap { $0 }
+    }
+
+    func parallelMap<T>(transform: @escaping ((Element) -> T)) -> [T] {
+        let count = self.count
+        let maxConcurrentJobs = ProcessInfo.processInfo.activeProcessorCount
+
+        guard count > 1 && maxConcurrentJobs > 1 else {
+            // skip GCD overhead if we'd only run one at a time anyway
+            return map(transform)
+        }
+
+        var result = [(Int, [T])]()
+        result.reserveCapacity(count)
+        let group = DispatchGroup()
+        let uuid = NSUUID().uuidString
+        let jobCount = Int(ceil(Double(count) / Double(maxConcurrentJobs)))
+
+        let queueLabelPrefix = "io.realm.SwiftLintFramework.map.\(uuid)"
```

Comment:
curious whether there is a reason you didn't use `DispatchQueue.concurrentPerform` here?